src/util/isShortcutExecutable.ts

```diff
@@ -0,0 +1,5 @@
+import { State } from '../util/initialState'
```

Comment:
Created a common util that can be reused rather than having the same logic in multiple files. This way if needed to modify in future, it will just one file change.

Reply:
Hmmm.... The places that I have modified does the same check. Hence I moved it to common file. If any shortcut does a different validation based on `isDocumentEditable` and `cursor`, I haven't touched those. Do you still think I should revert the changes and have the validation in respective shortcut file?

Reply:
Yes, reverting would be best.

To offer a little more explanation: Your approach is functionally correct, but ill-advised. I don't think colocation of code is a sufficient criterion for abstraction. In this case, grouping them into a function would suggest that they belong together, rather than that they are together for ad hoc reasons and fully independent. Hope that helps!

Reply:
Done.